# --- ROCK-COPYRIGHT-NOTE-BEGIN ---
# 
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# Please add additional copyright information _after_ the line containing
# the ROCK-COPYRIGHT-NOTE-END tag. Otherwise it might get removed by
# the ./scripts/Create-CopyPatch script. Do not edit this copyright text!
# 
# ROCK Linux: rock-src/package/base/kbd/kbdrate-nonx86.patch
# ROCK Linux is Copyright (C) 1998 - 2003 Clifford Wolf
# 
# This program is free software; you can redistribute it and/or modify
# it under the terms of the GNU General Public License as published by
# the Free Software Foundation; either version 2 of the License, or
# (at your option) any later version. A copy of the GNU General Public
# License can be found at Documentation/COPYING.
# 
# Many people helped and are helping developing ROCK Linux. Please
# have a look at http://www.rocklinux.org/ and the Documentation/TEAM
# file for details.
# 
# --- ROCK-COPYRIGHT-NOTE-END ---

--- kbd-1.08/src/kbdrate.c	2002-10-11 13:09:01.000000000 +0200
+++ kbd-1.08-fixed/src/kbdrate.c	2003-07-19 11:03:05.000000000 +0200
@@ -237,6 +237,7 @@
 	if(KIOCSRATE_ioctl_ok(rate, delay, silent))	/* sparc? */
 		return 0;
 
+#ifdef __i386__ 
 
 	/* The ioport way */
 
@@ -286,4 +287,8 @@
 			valid_delays[ (value & 0x60) >> 5 ] );
 
 	return 0;
+#else
+	perror( _("No alternative method for this platform") );
+	exit( 1 );
+#endif
 }
